### PR TITLE
Ci warnint v2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
None

Describe changes:
- Makes CI error on integer warnings !
- Fixes one newly introduced integer warning (by eb155639c6293e5fd412a8268a2c1a6ca03ebe45)

libhtp-pr: 369

Modifies https://github.com/OISF/suricata/pull/8136 by completeness with green CI

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 128237 | 103892 | 81.02% |
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 3060511 | - |
| .flow.end.state.new | 6272 | 24851 | 396.22% |
| .flow.end.tcp_state.syn_sent | 0 | 5780 | - |
| .flow.end.tcp_state.syn_recv | 0 | 44 | - |
| .flow.end.tcp_state.established | 53996 | 65562 | 121.42% |
| .flow.end.tcp_state.fin_wait1 | 0 | 782 | - |
| .flow.end.tcp_state.fin_wait2 | 0 | 233 | - |
| .flow.end.tcp_state.time_wait | 0 | 367 | - |
| .flow.end.tcp_state.last_ack | 0 | 204 | - |
| .flow.end.tcp_state.close_wait | 0 | 964 | - |
| .flow.end.tcp_liberal | 80952 | 168401 | 208.03% |
| .tcp.reassembly_gap | 80952 | 386869 | 477.9% |
| .tcp.overlap | 0 | 24 | - |
| .app_layer.error.http.parser | 0 | 226 | - |
| .app_layer.error.ftp.gap | 0 | 34 | - |
| .app_layer.error.smtp.gap | 0 | 450 | - |
| .app_layer.error.tls.gap | 53852 | 130396 | 242.14% |
| .app_layer.error.dcerpc_tcp.parser | 0 | 181 | - |
| .ftp.memuse | 0 | 56 | - | 

Pipeline 10385

File: src/defrag.c [hunk 1]
```diff
@@ -431,10 +431,10 @@ Defrag6Reassemble(ThreadVars *tv, DefragTracker *tracker, Packet *p)
     }
     PKT_SET_SRC(rp, PKT_SRC_DEFRAG);
 
-    int unfragmentable_len = 0;
-    int fragmentable_offset = 0;
-    int fragmentable_len = 0;
-    int ip_hdr_offset = 0;
+    uint16_t unfragmentable_len = 0;
```
Comment (victorjulien) [inline src/defrag.c:434]: I'm a bit worried with these that we might overflow them. I don't see strict checks below. Did you review this?

Comment (catenacyber) [reply src/defrag.c:434]: Will look into adding debug validations...

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/8161

**Pull request closed without merging**